### PR TITLE
Add Tests for Lifetime Validation Using New Validation Model For SAML2

### LINKED ISSUE #2885: Implement and Test LifeTime in Saml Token Handlers Using New Token Validation Model
Add lifetime validation to new ValidateTokenAsync() flow in both SamlSecurityTokenHandler and Saml2SecurityTokenHandler.

### PR DESCRIPTION
**Pull request by FuPingFranco**
# Add Tests for Lifetime Validation Using New Validation Model For SAML2

## Description

Adding tests validating lifetime for new implementation of ValidateTokenAsync for Saml2SecurityTokenHandler 

Fixes #2906

### CODE REVIEW

File: src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2Conditions.cs [hunk 1]
```diff
@@ -82,12 +82,12 @@ public DateTime? NotOnOrAfter
             {
                 value = DateTimeUtil.ToUniversalTime(value);
 
-                // NotBefore must be earlier than NotOnOrAfter
-                if (value != null && NotBefore.HasValue)
-                {
-                    if (value.Value <= NotBefore.Value)
-                        throw LogExceptionMessage(new ArgumentException(FormatInvariant(LogMessages.IDX13514, MarkAsNonPII(value), MarkAsNonPII(NotBefore))));
-                }
+                //This should not be checked here, instead fail during validation of the token. Will remove this code once we release new validation model bug #2905
```
Comment (iNinja) [inline src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2Conditions.cs:85]: nit: I usually find more clear the single line `//` comments as they are easier to know where they begin and end when looking at git history or PRs.

**Review by iNinja — APPROVED** [review]
LGTM

File: src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2Conditions.cs (hunk 1 above)
Comment (jennyf19) [reply src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2Conditions.cs:85]: In bug #2905 can you make sure this is linked there?

File: test/Microsoft.IdentityModel.Tokens.Saml.Tests/Saml2SecurityTokenHandlerTests.ValidateTokenAsyncTests.Lifetime.cs [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+                theoryData.Expires);
+
+            var tokenValidationParameters = CreateTokenValidationParameters(
+                saml2Token,
+                theoryData.NullTokenValidationParameters,
+                theoryData.ClockSkew);
+
+            Saml2SecurityTokenHandler saml2TokenHandler = new Saml2SecurityTokenHandler();
+
+            // Validate token using TokenValidationParameters
```
Comment (jennyf19) [inline test/Microsoft.IdentityModel.Tokens.Saml.Tests/Saml2SecurityTokenHandlerTests.ValidateTokenAsyncTests.Lifetime.cs:33]:
```suggestion
            // Validate token using legacy TokenValidationParameters
```

**Review by jennyf19 — APPROVED** [review]
:shipit:

**Review by brentschmaltz — APPROVED** [review]

**Pull request merged**